experiments/krico/workloads/caching.go

```
@@ -0,0 +1,218 @@
+	StartSnapService(workloadExecutorConfig.Hypervisor.Address)
+
+	//	Obtain workload instance cgroup.
+	cgroup, err := GetInstanceCgroup(workloadExecutorConfig.Hypervisor.InstanceName, workloadExecutorConfig.Hypervisor.Address)
+	errutil.CheckWithContext(err, "Cannot obtain instance cgroup!")
+
+	//
+	//	KRICO parameters
+	//
+
+	//	Calculate workload parameters.
+	memory := float64(workloadConfig.MaxMemoryMB / 1024)             // total cache size [GiB]
+	ratio, err := strconv.ParseFloat(loadGeneratorConfig.Update, 64) // estimated get vs set ratio [0.0 - 1.0]
+	errutil.CheckWithContext(err, "Cannot parse mutilate update argument (ratio parameter)!")
+	clients := float64(loadGeneratorConfig.MasterThreads * loadGeneratorConfig.MasterConnections)
+
+	//
+	//	Snap task
+	//
+
+	//	Configure task.
+	snapTaskConfig := kricosnapsession.DefaultConfig(cgroup, workloadExecutorConfig.Hypervisor.InstanceName)
+	snapTaskConfig.Tags = PrepareDefaultKricoTags(workloadExecutorConfig, experimentID)
+	snapTaskConfig.Tags["category"] = TypeCaching
+	snapTaskConfig.Tags["memory"] = strconv.FormatFloat(memory, 'f', -1, 64)
+	snapTaskConfig.Tags["ratio"] = strconv.FormatFloat(ratio, 'f', -1, 64)
+	snapTaskConfig.Tags["clients"] = strconv.FormatFloat(clients, 'f', -1, 64)
+
+	//	Prepare launcher.
+	snapTaskLauncher, err := kricosnapsession.NewSessionLauncher(snapTaskConfig)
+	errutil.CheckWithContext(err, "Cannot obtain Snap Task Launcher !")
+
+	//	Run Snap task.
+	snapTaskHandle, err := snapTaskLauncher.Launch()
+	errutil.CheckWithContext(err, "Cannot gather performance metrics!")
+
+	//	Stop on the end.
+	defer snapTaskHandle.Stop()
+
+	//
+	//	Load generator
+	//
+
+	//	Start load on workload.
+	loadGeneratorHandle, err := loadGeneratorLauncher.Load(maxQPS, loadDuration)
+	errutil.CheckWithContext(err, "Cannot start Mutilate task!")
+
+	//	In the end stop load on workload.
+	defer loadGeneratorHandle.Stop()
+
+	//	Wait until load generating finishes.
+	loadGeneratorHandle.Wait(0)
```

Do not ignore errors, please.

---

Error is still being ignored.